components/homme/src/theta-l_kokkos/cxx/DirkFunctorImpl.hpp

```diff
@@ -382,9 +386,10 @@ struct DirkFunctorImpl {
       } // Newton iteration
       kv.team_barrier();
 
-      if (it>=maxiter) {
-        printf ("[DIRK] WARNING! Newton reached max iteration count,"
-                " with deltaerr = %3.17f\n", deltaerr);
+      if (it >= maxiter) {
```

Comment:
Shouldn't we also check that `deltaerr>=deltatol`? Reaching convergence at the last iteration should not raise an error, no?

Reply:
the warning doesn't actually say its not converging, only that we reached maxiter :-)

Reply:
Right, but setting `nerr = 1` will later trigger an abort.

Reply:
I agree with Mark. Also, this is from the F90; I just reformatted the code here.

Reply:
It triggers the check routine to be called. The check routine may or may not abort, depending on the actual values of the state variables.

Reply:
So if the states are valid, it won't abort, even if it did not converge, correct?

Reply:
Correct.